pkg/controller/pxc/upgrade.go

```diff
@@ -477,6 +477,11 @@ func (r *ReconcilePerconaXtraDBCluster) getSecretHash(cr *api.PerconaXtraDBClust
 		return "", err
 	}
 
+	if secretName == cr.Spec.PXC.SSLSecretName || secretName == cr.Spec.PXC.SSLInternalSecretName{
+		if secretObj.Data["ca.crt"] == nil || secretObj.Data["tls.crt"] == nil || secretObj.Data["tls.key"] == nil {
+			return "", errors.New("one or more tls certificates are not found in secret")
+		}
+	}
```

Comment:
I would move this code to separate func. The reason is that this code break single responsibility principle. I mean function was written to get hashes but now it tries to validate particular certificates.

Reply:
kk, i'll rewrite this part with separate pr